Add Flatpak workflow procedures for Satellite (#3624)
theforeman/foreman-documentation: guides/common/modules/proc_installing-flatpak-applications.adoc

[id="installing-flatpak-applications_{context}"]
= Installing Flatpak applications

This procedure installs selected applications from the enabled Flatpak remotes on the command line.

.Prerequisites
* Flatpak is installed.
* The Red Hat Flatpak remote is enabled.

.Procedure

> . Run the following command on the managed host to add the {Project} as a Flatpak remote:
> +
> [options="nowrap", subs="+quotes,verbatim,attributes"]
> ----
> $ flatpak remote-add --authenticator-name=org.flatpak.Authenticator.Oci katello oci+https://satellite.example.com/
> ----

Reviewer: ```suggestion
. On the managed host, add your {ProjectServer} as a Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ flatpak remote-add --authenticator-name=org.flatpak.Authenticator.Oci katello oci+https://{foreman-example-com}/
----
```

[id="installing-flatpak-applications_{context}"]
= Installing Flatpak applications

This procedure installs selected applications from the enabled Flatpak remotes on the command line.

.Prerequisites
* Flatpak is installed.
* The Red Hat Flatpak remote is enabled.

.Procedure
. Run the following command on the managed host to add the {Project} as a Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ flatpak --user remote-add --authenticator-name=org.flatpak.Authenticator.Oci katello oci+https://satellite.example.com/
----
. Log in using podman:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ podman login
----
. Install the Mozilla Firefox Flatpak:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ flatpak install firefox
----

.Procedure using remote execution templates

> . Set up Flatpak remote on host:

Reviewer 2: Is this clear enough to say these 3 are remote templates? 
Set up Flatpak remote on host
Install Flatpak application
Log in to Flatpak registry using podman

I'd add a brief description of the 3. 

1st runs a REX job on host to add satellite/capsule as a flatpak remote on the host and logs into podman registry and saves the credentials.
2nd runs a REX job to install Flatpak applications on selected host(s)
3rd runs a REX job to login to podman registry and save credentials on the client.

Author: @sjha4 The more I try to clarify this with what you want me to add, the more I'm not sure we are telling the user what to do here. Should we keep this section using REX templates?

Reviewer 2: There are 3 templates admins can use to set up the clients. We should keep it for admins to be able to do what we document right above this using flatpak cli commands..We should even move it above the regular procedure in this section.

Reviewer 2: Updated REX templates to be better grouped for UX and follow convention. These should make it clearer in the documentation. 

**Flatpak - Set up remote on host
Flatpak - Install application on host
Flatpak - Login to registry via podman**

> * A user-provided name for the {Project} or {SmartProxy} remote.

Reviewer 2: One more required param here for the 1st REX template is :
Flatpak registry URL: Satellite_URL for satellite, capsule_url/pulpcore_registry/ for clients getting content from capsules.

[id="installing-flatpak-applications_{context}"]
= Installing Flatpak applications

This procedure installs selected applications from the enabled Flatpak remotes on the command line.

.Prerequisites
* Flatpak is installed.
* The Red Hat Flatpak remote is enabled.

.Procedure
. Run the following command on the managed host to add the {Project} as a Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> $ flatpak --user remote-add --authenticator-name=org.flatpak.Authenticator.Oci katello oci+https://satellite.example.com/

Reviewer 2: Lets remove --user from here and add a note for context if needed. The --user flag sets up remote and also installs apps in user context on the client. System-wide installs are the default and the REX templates also follow that.
